server/src/test/java/org/elasticsearch/common/cli/EnvironmentAwareCommandTests.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.cli;
+
+import joptsimple.OptionSet;
+
+import org.elasticsearch.Build;
+import org.elasticsearch.cli.Command;
+import org.elasticsearch.cli.CommandTestCase;
+import org.elasticsearch.cli.Terminal;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.env.Environment;
+import org.junit.Before;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+public class EnvironmentAwareCommandTests extends CommandTestCase {
```

Comment:
Can we also please replicate `DockerTests.test087EnvironmentVariablesInIncorrectFormatAreIgnored()`? It checks that env vars, which look like maybe they ought to be respected, are ignored due to being malformed.

You'll also need to remove these tests from `DockerTests`, because they rely on being able to inspect the Elasticsearch command line for options:

   * `test086EnvironmentVariablesInSnakeCaseAreTranslated`
   * `test087EnvironmentVariablesInIncorrectFormatAreIgnored`
   * `test087EnvironmentVariablesInIncorrectFormatAreIgnored`